graph/push.go

```diff
@@ -442,29 +443,22 @@ func (s *TagStore) pushV2Image(r *registry.Session, img *image.Image, endpoint *
 		os.Remove(tf.Name())
 	}()
 
-	ts, err := tarsum.NewTarSum(arch, true, tarsum.Version1)
+	h := sha256.New()
+	size, err := bufferToFile(tf, io.TeeReader(arch, h))
```

Comment:
Can we? yes, Should we? probably, but can we make a separate PR for it.  It will require additional testing and verifying with the registry that there are no problems.  We have no been using gzip on the engine side for the v2 registry thus far, but agree now is probably the best time to start.